src/components/pages/Home/Home.tsx

```diff
@@ -29,29 +36,73 @@ function Home() {
       }),
     );
     setFeaturedDomains(featuredDomains);
-  }, [arnsSourceContract]);
-  return (
-    <div className="page">
-      <div className="page-header h1">Arweave Name System</div>
-      <SearchBar
-        setIsSearching={setIsSearching}
-        values={records}
-        successPredicate={(value: string | undefined) =>
-          isArNSDomainNameAvailable({ name: value, records })
-        }
-        validationPredicate={(value: string | undefined) =>
-          isArNSDomainNameValid({ name: value })
-        }
-        placeholderText={'Enter a name'}
-        headerElement={<SearchBarHeader defaultText={'Find a domain name'} />}
-        footerElement={
-          <SearchBarFooter
-            defaultText={
-              'Names must be 1-32 characters. Dashes are permitted, but cannot be trailing characters and cannot be used in single character domains.'
+    setRegistrationWorkflow({
+      0: {
+        component: (
+          <SearchBar
+            values={records}
+            successPredicate={(value: string | undefined) =>
+              isArNSDomainNameAvailable({ name: value, records })
+            }
+            validationPredicate={(value: string | undefined) =>
+              isArNSDomainNameValid({ name: value })
+            }
+            placeholderText={'Enter a name'}
+            headerElement={
+              <SearchBarHeader defaultText={'Find a domain name'} />
             }
+            footerElement={
+              <SearchBarFooter
+                defaultText={
+                  'Names must be 1-32 characters. Dashes are permitted, but cannot be trailing characters and cannot be used in single character domains.'
+                }
+              />
+            }
+            height={45}
           />
-        }
-      />
+        ),
+        nextCondition: true,
+        backCondition: true,
```

Comment:
what is a use case for `backCondition`?

Reply:
For when you dont want the user to be able to go back, or when you have an alternative display component or conditional that handles going back a step. For example, during deployment have a transaction, it could be found that a user suddenly has insufficient funds for gas or payment, the current comp can then handle the back condition. Currently this bool simply handles the actual rendering of the back button instead of the condition of the back button being available. A better solution would be to come back in another pr and advance this to a predicate for both the nextCondition and backCondition - however with the current pattern that is difficult to lay out because of the components being initialized outside of the workflow component - maybe we can use some sort of generic even emitter on the workflow buttons where they know "ok component "search" emitted its onSubmit state, ()=> predicate based on constants from nextCondition... like arg0 === false/string with x chars or something.

Reply:
i understand the intent, but still circling my head around an ideal implementation. fine to leave for now and address in another PR